libexec/functions

```diff
@@ -396,30 +396,12 @@ nonroot_build_warning() {
 is_deffile() {
 # check if a file looks like, walks like, and quacks like a def file
     FILE2CHECK=$1
-    
-    # first make a guess based on naming convention
-    case ${FILE2CHECK:-} in
-        *.img)
-            return 1
-            ;;
-        *.tar*)
-            return 1
-            ;;
-        *.def)
-            return 0
-            ;;
-        Singularity) 
-            return 0
-            ;;
-        *)
-            ;;
-        esac
 
-    # name is ambiguous.  is it a text file with def file words in it?
+    # is it a text file with def file words in it?
     if ! `file $FILE2CHECK | grep -q "ASCII\|Unicode"`; then
```

Comment:
What about grepping the output of `strings`?  Looks like `strings` is included in alpine and busybox.

Reply:
On second thought, I don't think that is going to work b/c strings will happily pull those keywords out of a def file that is stored in an image.  😕 

I actually think we tried that when we were first writing these tests and had problems.

Reply:
I can't think of a nice way to do this if file isn't available. I think using the file check first, and dropping to the less-smart name check if it isn't is probably the sensible thing. When we are in Go in 3.0 we will have an avenue to address this better.

Reply:
meh.  Is it really worth it to make the change if we are just going to reverse the order of the checks and retain the file-name business?  Especially if this will be a moot point once 3.0 comes out?

Reply:
Probably not - there is a straightforward workaround in file naming here.